docs/descriptor.rst

```diff
@@ -366,11 +371,45 @@ Home directory `~` syntax will be expanded:
 
     sgtk:descriptor:path?path=%7E/path/to/app
 
-Sometimes it can be useful to organize your development sandbox relative to a pipeline configuration.
-If all developers in the studio share a convention where they for example have a ``dev`` folder inside
-their pipeline configuration dev sandboxes, it becomes easy to exchange environment configs.
-You can achieve this by using the special token ``{PIPELINE_CONFIG}`` which will resolve into the
-local path to the pipeline configuration:
+
+Sometimes it may be useful to distribute application code within the configuration for applications
+that may not be accessible through the traditional distribution mechanism for Toolkit applications
+like the :ref:`App Store<app_store_descriptor>` or the various :ref:`Git<git_descriptors>`
+descriptor types. Although we recommend in those case to use the :ref:`local bundle
+caches<local_bundle_caches>`, it may not always be desirable. In those cases, it is possible to
+package the application code alongside the ``env`` or ``hooks`` folder and refer to it with the
+``CONFIG_FOLDER`` token:
+
+.. code-block:: yaml
+
+    {
+        type: dev,
+        path: {CONFIG_FOLDER}/dev/tk-nuke-myapp
+    }
+
+.. code-block:: yaml
+
+    sgtk:descriptor:dev?path=%7BCONFIG_FOLDER%7D/dev/tk-nuke-myapp
+
+
+When using an installed pipeline configuration, this will refer to
+``{PIPELINE_ROOT}/config``. When using a cached or baked pipeline configuration, it will point where
+the bundle was found in one of the bundle caches.
+
+.. image:: ./resources/descriptor/config_folder_token.png
```

Comment:
nitpick: if not too much trouble, maybe include the `def/tk-nuke-myapp` structure for both screenshots in the the image to help support the example above?